lib/custom-logger.js

```diff
@@ -26,15 +30,82 @@ function defaultRedactionRules() {
   }
 }
 
+function timestampFunction() {
+  return `,"time":${Date.now() / 1000.0}`
+}
+
 function pinoOptions(moduleOptions, options) {
+  const stream = options.stream ? { stream: options.stream } : {}
   return {
     level: options.logLevel || moduleOptions.logLevel || 'info',
     redact: moduleOptions.redact || defaultRedactionRules(),
+    ...stream,
+    serializers: {
+      req: requestSerializer,
+      res: responseSerializer,
+    },
+    timestamp: timestampFunction,
+  }
+}
+
+function requestSerializer(request) {
+  return {
+    http: {
+      request: {
+        method: request.method,
+      },
+    },
+    url: {
+      full: request.url,
+    },
+    userAgent: {
+      original: request.headers['user-agent'],
+    },
+    host: {
+      hostname: request.hostname,
+      ip: request.ip,
+    },
+  }
+}
+
+function responseSerializer(response) {
+  return {
+    http: {
+      response: {
+        statusCode: response.statusCode,
+        body: {
+          bytes: response.getHeader('content-length'),
+        },
+      },
+    },
   }
 }
 
 module.exports = function customLogger(moduleOptions, options) {
-  return pinoOptions(moduleOptions, options)
+  const { stream } = options
+  const transform = (logLine, enc, next) => {
+    const { req, res, ...transformedLog } = JSON.parse(logLine.toString())
+    next(null, `${JSON.stringify({
+      ...transformedLog,
+      ...(req || {}),
+      ...(res || {}),
+    })}\n`)
```

Comment:
uhm... this is a huge changing. JSON.parse and JSON.stringify is not performant.

Reply:
I know but we're still treating small text strings, I'm open to suggestions for better performances,

Anyhow before proceeding and optimize this transform function I'd prefer writing some benchmarks